### PR TITLE
Fix broken test + add samples for QueueSelectors in classification policy

### PR DESCRIPTION
**Pull request by sarkar-rajarshi**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.JobRouter/samples/Sample1_HelloWorld.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+## Import the namespaces
+
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_UsingStatements
+using Azure.Communication.JobRouter;
+using Azure.Communication.JobRouter.Models;
+```
+
+## Create a client
+
+Create a `RouterClient` and send a request.
```
Comment (danielgerlag) [inline sdk/communication/Azure.Communication.JobRouter/samples/Sample1_HelloWorld.md:12]: Do we need the "and send a request" bit?

File: sdk/communication/Azure.Communication.JobRouter/samples/Sample1_HelloWorld.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+Use `RouterClient` to create a [Distribution Policy](https://docs.microsoft.com/en-us/azure/communication-services/concepts/router/concepts#distribution-policy) to control how jobs are to be distributed to workers with associated queue.
+
+For this example, we are going to create a __Longest Idle__ policy with an offer TTL set to 30 seconds.
+
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_CreateDistributionPolicyLongestIdleTTL30S
+var distributionPolicy = routerClient.SetDistributionPolicy(
+    id: "distribution-policy-1",
+    name: "My Distribution Policy",
+    offerTTL: TimeSpan.FromSeconds(30),
```
Comment (danielgerlag) [inline sdk/communication/Azure.Communication.JobRouter/samples/Sample1_HelloWorld.md:28]: Maybe, for the first sample we should have a longer offer TTL than 3 seconds, so developers experimenting with it have a chance to see the offer?

File: sdk/communication/Azure.Communication.JobRouter/samples/Sample2_ClassificationWithQueueSelector.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
+# Azure.Communication.JobRouter Samples - Using QueueSelectors in Classification Policy
```
Comment (danielgerlag) [inline sdk/communication/Azure.Communication.JobRouter/samples/Sample2_ClassificationWithQueueSelector.md:1]: I preview 2, we've simplified the queue selector to only be a label selector, so I'm not sure if it's worth documenting the others.

File: sdk/communication/Azure.Communication.JobRouter/tests/Infrastructure/RouterLiveTestBase.cs [hunk 4]
```diff
@@ -61,7 +61,14 @@ protected void AssertQueueResponseIsEqual(Response<UpsertQueueResponse> upsertQu
             Assert.AreEqual(distributionPolicyId, response.DistributionPolicyId);
             if (queueLabels != default)
             {
-                Assert.AreEqual(queueLabels, response.Labels);
+                var someting = queueLabels.ToDictionary(k => k.Key, k => k.Value);
+                Assert.True(response.Labels.ContainsKey("Id"));
+                Assert.AreEqual(queueId, response.Labels["Id"]);
+
+                // copy label collection to a new dictionary - do not implicitly modify response object within function
+                var responseLabelCollection =
```
Comment (danielgerlag) [inline sdk/communication/Azure.Communication.JobRouter/tests/Infrastructure/RouterLiveTestBase.cs:69]: Would it make sense to have a reusable extension method or something for label comparison for tests?

File: sdk/communication/Azure.Communication.JobRouter/samples/Sample1_HelloWorld.md (hunk 1 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/samples/Sample1_HelloWorld.md]: removed

File: sdk/communication/Azure.Communication.JobRouter/samples/Sample1_HelloWorld.md (hunk 2 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/samples/Sample1_HelloWorld.md]: changed to have TTL of 1Day

File: sdk/communication/Azure.Communication.JobRouter/samples/Sample2_ClassificationWithQueueSelector.md (hunk 3 above)
Comment (sarkar-rajarshi) [reply sdk/communication/Azure.Communication.JobRouter/samples/Sample2_ClassificationWithQueueSelector.md:1]: removed QueueIdSelector

**Review by NathanJBennett — APPROVED** [review]

**Pull request merged**